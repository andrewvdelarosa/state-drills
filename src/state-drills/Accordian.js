import React from 'react';


export default class Accordian extends React.Component {
    static defaultProps = {
        sections: []
    };
        
    state = {
        sectionIndex: null,
    }

     handleClick = (sectionIdx) => {
        this.setState({ sectionIndex: sectionIdx })
      }

      displayResults(section, idx, sectionIndex) {
         return (
                <li key={idx} className="item">
                    <button 
                    type="button" 
                    onClick={() => this.handleClick(idx)}
                    >
                        {section.title}
                    </button>
                    {(sectionIndex === idx) && <p>{section.content}</p>}
                </li>
            )
      }
    

      render() {
          const { sectionIndex } = this.state
          const { sections } = this.props
          return (
              <ul className="Accordian">
                  {sections.map((section, idx) => 
                     this.displayResults(section, idx, sectionIndex)
                    )
                }
              </ul>
          )
        }
}