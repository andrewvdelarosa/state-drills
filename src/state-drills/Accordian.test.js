import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Enzyme from "enzyme";
import Accordian from './Accordian'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() });


const sections = [{
  title: "Section 1",
  content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
},
{
  title: "Section 2",
  content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
},
{
  title: "Section 3",
  content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
},]


describe(`accordian component`, () => {

  it("Accordian renders", () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it("Renders the section by default", () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("Opens a click section", () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find("button").at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("Opens a section at a time", () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find("button").at(1).simulate('click')
    wrapper.find("button").at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})

