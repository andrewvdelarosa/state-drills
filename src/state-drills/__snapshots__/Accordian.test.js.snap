// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordian component Accordian renders 1`] = `
<ul
  className="Accordian"
/>
`;

exports[`accordian component Opens a click section 1`] = `
<ul
  className="Accordian"
>
  <li
    className="item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`accordian component Opens a section at a time 1`] = `
<ul
  className="Accordian"
>
  <li
    className="item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`accordian component Renders the section by default 1`] = `
<ul
  className="Accordian"
>
  <li
    className="item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    className="item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    className="item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
